File: C++/Sorting/selection_sort.cpp
#include<bits/stdc++.h>
#include<iostream>
using namespace std;


void selection_sort(int arr[], int n){
	int i, j, k;
	
	for(i = 0; i < n-1; i++){
		k = i;
		for(j = i+1; j < n; j++)
		if(arr[j] < arr[k])
			k = j;
			
		swap(arr[k], arr[i]);
			
		
	}

}

int main(){
	int arr[] = {53, 123, 22, 20, 1, 223, 15, 4141};
	int n = sizeof(arr)/sizeof(arr[0]);
	
	selection_sort(arr, n);
	
	printf("The sorted array: \n");
	for(int i = 0; i < n; i++){
		printf(" %d ,", arr[i]); 
	}
}

File: Ruby/Sorting/SelectionSort.rb
def selectionSort(arr)
    n = arr.length
   for i in 0...n 
    min = i
    for j in (i+1)...n
        if arr[j]<arr[min]
            temp=arr[j]
            arr[j]=arr[min]
            arr[min]=temp
        end
    end
   end
   return arr
end


array = [1, -4, 6, 11, -6, 2, 78, 32, 48, 21]
sorted = selectionSort(array)
print(sorted)